io/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+pub enum Move {
+    Rock,
+    Paper,
+    Scissors,
+    Lizard,
+    Spock,
+}
+
+impl Move {
+    pub fn number(&self) -> char {
+        match self {
+            Move::Rock => '0',
+            Move::Paper => '1',
+            Move::Scissors => '2',
+            Move::Lizard => '3',
+            Move::Spock => '4',
+        }
+    }
+
+    pub fn new(number: char) -> Move {
+        match number {
+            '0' => Move::Rock,
+            '1' => Move::Paper,
+            '2' => Move::Scissors,
+            '3' => Move::Lizard,
+            '4' => Move::Spock,
+            _ => panic!("Unknown symbol in move"),
+        }
+    }
+
+    pub fn wins(&self, other: &Move) -> bool {
+        match self {
+            Move::Rock => match other {
+                Move::Rock | Move::Paper | Move::Spock => false,
+                Move::Scissors | Move::Lizard => true,
+            },
+            Move::Paper => match other {
+                Move::Paper | Move::Scissors | Move::Lizard => false,
+                Move::Rock | Move::Spock => true,
+            },
+            Move::Scissors => match other {
+                Move::Rock | Move::Scissors | Move::Spock => false,
+                Move::Paper | Move::Lizard => true,
+            },
+            Move::Lizard => match other {
+                Move::Rock | Move::Scissors | Move::Lizard => false,
+                Move::Paper | Move::Spock => true,
+            },
+            Move::Spock => match other {
+                Move::Paper | Move::Lizard | Move::Spock => false,
+                Move::Rock | Move::Scissors => true,
+            },
+        }
+    }
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub struct StageDescription {
+    pub anticipated_players: BTreeSet<ActorId>,
+    pub finished_players: BTreeSet<ActorId>,
+}
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub enum GameStage {
+    #[default]
+    Preparation,
+    InProgress(StageDescription),
+    Reveal(StageDescription),
+}
+
+impl GameStage {
+    pub fn game_is_in_progress(&self) -> bool {
+        match self {
+            GameStage::Preparation => false,
+            GameStage::InProgress(_) | GameStage::Reveal(_) => true,
+        }
+    }
+
+    pub fn move_can_be_made(&self) -> bool {
+        match self {
+            GameStage::Preparation | GameStage::InProgress(_) => true,
+            GameStage::Reveal(_) => false,
+        }
+    }
+
+    pub fn is_player_in_game(&self, player: &ActorId) -> bool {
+        match self {
+            GameStage::Preparation => false,
+            GameStage::InProgress(description) => {
+                description.anticipated_players.contains(player)
+                    || description.finished_players.contains(player)
+            }
+            GameStage::Reveal(description) => {
+                description.anticipated_players.contains(player)
+                    || description.finished_players.contains(player)
+            }
```

Comment:
Can be combined for the sake of shortness.
```suggestion
            GameStage::InProgress(description) | GameStage::Reveal(description) => {
                description.anticipated_players.contains(player)
                    || description.finished_players.contains(player)
            }
```